java-extensions/jni-connector/src/main/java/com/starrocks/jni/connector/OffHeapColumnVector.java

```diff
@@ -345,11 +335,240 @@ public String getUTF8String(int rowId) {
+        }
+        return;
+    }
+
+    public void appendValue(ColumnValue o) {
+        // TODO(yanz): FIXME.
+        ColumnType.TypeValue typeValue = type.getTypeValue();
+        if (o == null) {
+            appendNull();
+            // NOTE(yan): for struct need to fill
+            if (type.isStruct()) {
+                List<ColumnValue> values = new ArrayList<>();
+                for (int i = 0; i < type.childTypes.size(); i++) {
+                    values.add(null);
+                }
+                appendStruct(values);
+            }
+            return;
+        }
+
+        switch (typeValue) {
+            case BOOLEAN:
+                appendBoolean(o.getBoolean());
+                break;
+            case SHORT:
+                appendShort(o.getShort());
+                break;
+            case INT:
+                appendInt(o.getInt());
+                break;
+            case FLOAT:
+                appendFloat(o.getFloat());
+                break;
+            case LONG:
+                appendLong(o.getLong());
+                break;
+            case DOUBLE:
+                appendDouble(o.getDouble());
+                break;
+            case STRING:
+            case DATE:
+            case DECIMAL:
+                appendString(o.getString());
+                break;
+            case ARRAY: {
+                List<ColumnValue> values = new ArrayList<>();
+                o.unpackArray(values);
+                appendArray(values);
+                break;
+            }
+            case MAP: {
+                List<ColumnValue> keys = new ArrayList<>();
+                List<ColumnValue> values = new ArrayList<>();
+                o.unpackMap(keys, values);
+                appendMap(keys, values);
+                break;
+            }
+            case STRUCT: {
+                List<ColumnValue> values = new ArrayList<>();
+                o.unpackStruct(type.getStructFieldIndex(), values);
+                appendStruct(values);
+                break;
+            }
+            default:
+                throw new RuntimeException("Unknown type value: " + typeValue);
+        }
+        return;
+    }
+
+    public void checkMeta(OffHeapTable.MetaChecker checker) {
+        checker.check(nulls);
+        if (type.isString()) {
+            checker.check(arrayOffsetNativeAddress());
+            checker.check(arrayDataNativeAddress());
+        } else if (type.isArray()) {
+            checker.check(arrayOffsetNativeAddress());
+            childColumns[0].checkMeta(checker);
+        } else if (type.isMap()) {
+            checker.check(arrayOffsetNativeAddress());
+            childColumns[0].checkMeta(checker);
+            childColumns[1].checkMeta(checker);
+        } else if (type.isStruct()) {
+            for (OffHeapColumnVector c : childColumns) {
+                c.checkMeta(checker);
+            }
+        } else {
+            checker.check(data);
+        }
```

Comment:
add comments of test only

The function name here is confusing with `OffheapTable::checkMeta` and we should distinguish these for readability. 
For example, change `OffheabTable::checkMeta`  to `OffheabTable::checkTableMeta`